ShortestPath/ShortestPath/ArrayHeap.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ShortestPath
+{
+    public class ArrayHeap : IHeap
+    {
+        public List<Node> Heap = new List<Node>();
+        private int count;
+
+        public Node Root => Heap[0];
+
+        public ArrayHeap(Dictionary<string, Node> nodes)
+        {
+            count = 0;
+            foreach (KeyValuePair<string, Node> node in nodes)
+            {
+                Add(node.Value);
+            }
+        }
+
+        public ArrayHeap()
+        {
+        }
+
+        public void Add(Node node)
+        {
+            Heap.Add(node);
+            count++;
+            PerculateUp();
+        }
+
+        private void PerculateUp()
+        {
+            if (Heap.Count == 0) { return; }
+
+            else
+            {
+                int pointer = count - 1;
+                while (true)
+                {
+                    Node Child = Heap[pointer];
+                    Node Parent = Heap[Math.Abs((pointer) / 2)]; //Maybe need to round down 
+                    Node comparer;
+                    if (Child.Cost < Parent.Cost)
+                    {
+                        comparer = Child;
+                        Heap[pointer] = Heap[Math.Abs((pointer) / 2)];
+                        Heap[Math.Abs((pointer) / 2)] = comparer;
+                    }
+                    pointer = Math.Abs((pointer) / 2);
+                    if (pointer == 0) { break; }
+                }
+            }
+        }
+        public Node Remove()
+        {
+            Node Root = Heap[0];
+            Heap[0] = Heap[count - 1];
+            Heap.RemoveAt(count - 1);
+            count--;
+            PerculateDown();
+            return Root;
+        }
+
+        private void PerculateDown()
+        {
+            int pointer = 0;
+            while (true)
+            {
+                if (Heap.Count == 0) { break; }
+
+                Node Parent = Heap[pointer];
+                Node LChild = null;
+                Node RChild = null;
+
+                if ((2 * pointer + 1 <= count - 1)) { LChild = Heap[2 * pointer + 1]; }
```

Comment:
please double check the arithmetic here, look at my other comment about line 47 (I didn't check myself, it can be right)